core/vm/interface.go

```diff
@@ -82,16 +83,20 @@ type StateDB interface {
 	// new methods
 	Error() error
 	GetStorageRoot(common.Address) common.Hash
-	SetBalance(common.Address, *big.Int)
+	SetBalance(common.Address, *big.Int, state.BalanceChangeReason)
 	SetStorage(common.Address, map[common.Hash]common.Hash)
 	Finalise(deleteEmptyObjects bool)
 	Commit(block uint64, deleteEmptyObjects bool) (common.Hash, error)
 	SetTxContext(thash common.Hash, ti int)
-	Copy() StateDB
+	Copy() interface{}
```

Comment:
I had to change the interface a bit. I had to remove type `vm.XXX` like the `Copy() StateDB` here as well as `SetEVM(evm *EVM)`.

Indeed, the new tracer in Geth code brings in `go-ethereum/core/state` but also adds `go-ethereum/core/state` inside `go-ethereum/core/vm`.

The problem that is was causing is that on the `StateDB` struct where `vm.StateDB` is implemented implicitly, with the Sei chain, `go-ethereum/core/state/statedb.go` is now in the obligation of pulling `go-ethereum/core/vm` so it can properly implement `Copy` and `SetEVM`. But this caused an import cycle now because `go-ethereum/core/vm` depends of ``go-ethereum/core/state` now.

So, I kept the copy but returned an `interface{}` and cast to `vm.StateDB` only when needed which breaks the dependency on `vm.StateDB`.

As for `SetEVM`, since this is actually used by the Sei `DBImpl` struct, I removed it from the interface so that Geth `StateDB` struct doesn't have to implement it.

Then you can see https://github.com/sei-protocol/go-ethereum/pull/15/files#diff-66f4d55a856f5b04d4c6e1ee99582788b0510add7bd64c7332b948c2aeadbfdfR216-R218